src/pages/videoListing/VideoListing.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { useState, useEffect } from 'react'
+import Navbar from 'components/navbar/Navbar'
+import Footer from 'components/footer/Footer'
+import VideoCard from 'components/videocard/VideoCard'
+
+export default function VideoListing() {
+  const [videos, setVideos] = useState([])
+  useEffect(() => {
+    ;(async () => {
+      const response = await fetch('/api/videos')
+      const data = await response.json()
+      setVideos(data.videos)
```

Comment:
Please use trycatch block, It will handle the error scenarios as well. You can show the error state in UI, so that UI looks good and will not break even if api fails.

Reply:
Thanks, updated the code.